src/vsr/replica.zig

```diff
@@ -4767,6 +4767,22 @@ pub fn ReplicaType(
             if (self.ignore_request_message_upgrade(message)) return true;
             if (self.ignore_request_message_duplicate(message)) return true;
             if (self.ignore_request_message_preparing(message)) return true;
+
+            // Only valid requests are accepted.
+            if (StateMachine.operation_from_vsr(message.header.operation)) |operation| {
+                if (!StateMachine.input_valid(operation, message.body())) {
+                    log.debug(
+                        "{}: on_request: ignoring invalid request (operation={s}, input_len={})",
+                        .{
+                            self.replica,
+                            @tagName(operation),
+                            message.body().len,
+                        },
+                    );
```

Comment:
A client that sent an invalid request would just endlessly retry. Maybe we could add a new eviction reason? Then at least the error surfaces. Likewise, this should probably be at least a `log.info` or maybe `log.warn`, since it is an exceptional case.

Reply:
I think it must behave similarly when the client sends an invalid operation.

- Let's log as `err` in both
- Should we evict the client in both cases??

```zig
            if (!message.header.operation.valid(StateMachine)) {
                // Some possible causes:
                // - client bug
                // - client memory corruption
                // - client/replica version mismatch
                log.err("{}: on_request: ignoring invalid operation (client={} operation={})", .{
                    self.replica,
                    message.header.client,
                    @intFromEnum(message.header.operation),
                });
                return true;
            }
```

Reply:
Yes, I think we should evict the client in both cases, since this is a severe/unexpected issue (probably a bug) and the session is unrecoverable. I would use a different eviction `Reason` for each.